api/lcp/collection.py

```diff
@@ -0,0 +1,311 @@
+        else:
+            license = self._lcp_server.generate_license(
+                self._db,
+                licensepool.identifier.identifier,
+                patron.id,
+                today,
+                expires
+            )
+
+            patron.external_identifier = license['encryption']['user_key']['key_check']
+
+        loan = LoanInfo(
+            licensepool.collection,
+            licensepool.data_source.name,
+            identifier_type=licensepool.identifier.type,
+            identifier=licensepool.identifier.identifier,
+            start_date=today,
+            end_date=expires,
+            fulfillment_info=None,
+            external_identifier=license['id']
+        )
+
+        return loan
+
+    def fulfill(self, patron, pin, licensepool, internal_format=None, part=None, fulfill_part_url=None):
+        """Get the actual resource file to the patron.
+
+        :param patron: A Patron object for the patron who wants to check out the book
+        :type patron: Patron
+
+        :param pin: The patron's alleged password
+        :type pin: string
+
+        :param licensepool: Contains lending info as well as link to parent Identifier
+        :type licensepool: LicensePool
+
+        :param internal_format: A vendor-specific name indicating the format requested by the patron
+        :type internal_format:
+
+        :param part: A vendor-specific identifier indicating that the
+            patron wants to fulfill one specific part of the book
+            (e.g. one chapter of an audiobook), not the whole thing
+        :type part: Any
+
+        :param fulfill_part_url: A function that takes one argument (a
+            vendor-specific part identifier) and returns the URL to use
+            when fulfilling that part
+        :type fulfill_part_url: Any
+
+        :return: a FulfillmentInfo object
+        :rtype: FulfillmentInfo
+        """
+        loan = get_one(self._db, Loan, patron=patron, license_pool=licensepool, on_multiple='interchangeable')
+        license = self._lcp_server.get_license(loan.external_identifier, loan.external_identifier)
+        fulfillment_info = LCPFulfilmentInfo(
+            licensepool.identifier.identifier,
+            licensepool.collection,
+            licensepool.data_source.name,
+            licensepool.identifier.type,
+            content_link=None,
+            content_type=DeliveryMechanism.LCP_DRM,
+            content=license,
+            content_expires=None,
+        )
+
+        return fulfillment_info
+
+    def patron_activity(self, patron, pin):
+        """Returns patron's loans
+
+        :param patron: A Patron object for the patron who wants to check out the book
+        :type patron: Patron
+
+        :param pin: The patron's alleged password
+        :type pin: string
+
+        :return: List of patron's loans
+        :rtype: List[LoanInfo]
+        """
+        # NOTE: We assume that there can be only ONE LCP collection per library
```

Comment:
If there were multiple LCP collections per library I'd expect each one to end up with a distinct `LCPAPI`. `patron_activity` would be called on each `LCPAPI` in parallel. So I don't think you're actually making this assumption.

Reply:
Removed the comment